TestFoundation/TestOperationQueue.swift

```diff
@@ -178,6 +180,51 @@ class TestOperationQueue : XCTestCase {
         
         waitForExpectations(timeout: 1)
     }
+
+    func test_cancelDependency() {
+        let expectation = self.expectation(description: "Operation should finish")
+
+        let queue = OperationQueue()
+        queue.maxConcurrentOperationCount = 6
+
+        let op1 = BlockOperation() {
+            XCTAssert(false, "Should not run")
+        }
+        let op2 = BlockOperation() {
+            expectation.fulfill()
+        }
+
+        op2.addDependency(op1)
+        op1.cancel()
+
+        queue.addOperation(op1)
+        queue.addOperation(op2)
+
+        waitForExpectations(timeout: 1)
+    }
+
+    func test_deadlock() {
+        let queue = OperationQueue()
+        queue.maxConcurrentOperationCount = 1
+
+        let expectation1 = self.expectation(description: "Operation should finish")
+        let expectation2 = self.expectation(description: "Operation should finish")
+
+        let op1 = BlockOperation {
+            expectation1.fulfill()
+        }
+        let op2 = BlockOperation {
+            expectation2.fulfill()
+        }
+
+        op1.addDependency(op2)
+
+        queue.addOperation(op1)
+        sleep(1)
+        queue.addOperation(op2)
+
+        waitForExpectations(timeout: 1)
```

Comment:
the sleep 1 might play havoc with the timeout here btw ( would suggest reducing the sleep to something smaller and that way the test is quicker too...)